src/response.rs

```diff
@@ -465,6 +539,21 @@ impl Response {
         self.body.is_empty()
     }
 
+    // XXX(Jeremiah) What should this be called?
+    #[cfg(feature = "unstable")]
+    /// Returns true if the response was created from an `Error`.
+    pub fn was_from_error(&self) -> bool {
+        self.error.is_some()
+    }
+
+    // XXX(Jeremiah) What should this be called?
+    #[cfg(feature = "unstable")]
+    /// Returns an `Error` if the response was created from one, or else `None`.
+    pub fn get_from_error(&mut self) -> Option<Error> {
```

Comment:
This and `was_from_error` could just be `Response::error(&self) -> Option<&Error>` (or `&Option<Error>`) and a `Response::take_error(&mut self) -> Option<Error>`

Reply:
Changed this to be `has_error() -> bool` and an `error() -> Option<Rc<Error>>` .

Reply:
Why an Rc? I think I don't understand the use case for when this would be accessed through an Rc that a reference wouldn't work. And isn't has_error identical to response.error().is_some()?

Reply:
True on further thought. Changed to:
- `error() -> Option<&Error>`
- `take_error() -> Option<Error>`